Alerting: Remove video from Alert Getting Started page

tomratcliffe opened the pull request:
**What is this feature?**

Removes the video from the Alerting getting started page, and organises the columns a little with this in mind

Before:
![before](https://github.com/grafana/grafana/assets/4377076/01c27d62-34e0-4f3c-a1f9-9ed3c5ad75ea)


After:
![after](https://github.com/grafana/grafana/assets/4377076/f213d01e-0629-4140-9c5b-65b970c31ae6)


**Why do we need this feature?**

As per https://github.com/grafana/grafana/issues/76020, tidying up the getting started page for alerting

**Which issue(s) does this PR fix?**:
Fixes #76020

**Special notes for your reviewer:**
I held off migrating everything to `@grafana/ui` components in this PR, as it would likely result in a number of other small changes to the UI that I didn't want to conflate with the simple change of removing the video.

tomratcliffe committed:
Remove video from Alert Getting Started page

tomratcliffe committed:
Remove unused `showWelcomeHeader` prop

tomratcliffe committed:
Tidy up styling for Getting Started page and use @grafana/ui components

tomratcliffe committed:
Convert getting started styles to object syntax

tomratcliffe commented on public/app/features/alerting/unified/home/GettingStarted.tsx:
```diff
@@ -20,16 +20,15 @@ export const getOverviewScene = () => {
 };
 
-export default function GettingStarted({ showWelcomeHeader }: { showWelcomeHeader?: boolean }) {
+export default function GettingStarted() {
```
`showWelcomeHeader` wasn't used anywhere else in the codebase from what I could tell

tomratcliffe commented on public/app/features/alerting/unified/home/GettingStarted.tsx:
```diff
@@ -266,26 +225,9 @@ function ContentBox({ children, className }: React.PropsWithChildren<{ className
 }
 
 const getContentBoxStyles = (theme: GrafanaTheme2) => ({
```
`ContentBox` is very similar in styling to the `@grafana/ui` Card component, but the heading of the Card isn't consistent with how we want to display it here, and switching over in this PR has a few minor knock-on effects/considerations

tomratcliffe commented on public/app/features/alerting/unified/home/GettingStarted.tsx:
```diff
@@ -266,26 +225,9 @@ function ContentBox({ children, className }: React.PropsWithChildren<{ className
-  `,
-});
-
-function ArrowLink({ href, title }: { href: string; title: string }) {
```
Replaced with the `TextLink` component instead

tomratcliffe commented on public/app/features/alerting/unified/home/GettingStarted.tsx:
```diff
@@ -216,47 +177,45 @@ function WelcomeCTABox({ title, description, href, hrefText }: WelcomeCTABoxProp
   return (
     <div className={styles.container}>
-      <h3 className={styles.title}>{title}</h3>
+      <Text element="h2" variant="h3">
```
Rendering as h2 otherwise we have a skipped heading level on the page

tomratcliffe committed:
Use Text component more consistently in GettingStarted

tomratcliffe commented on public/app/features/alerting/unified/home/GettingStarted.tsx:
```diff
@@ -182,26 +146,25 @@ const getWelcomeHeaderStyles = (theme: GrafanaTheme2) => ({
-    width: 1px;
-    background-color: ${theme.colors.border.medium};
+  ctaContainer: css({
+    padding: theme.spacing(2),
```
I changed the padding here slightly so that the welcome banner ("Alert rules, Contact points...") and "How it works" were both left aligned with the same spacing. If this was intentional originally then happy to revert :)

tomratcliffe commented on public/app/features/alerting/unified/home/GettingStarted.tsx:
```diff
@@ -182,26 +146,25 @@ const getWelcomeHeaderStyles = (theme: GrafanaTheme2) => ({
+      flexDirection: 'column',
+    },
+  }),
+  separator: css({
```
There is a `Divider` component in `@grafana/ui` but I tried using that instead and it needed other changes to make it behave in flex, and the spacing was off. Again, I didn't want to overdo it in this PR

gillesdemey commented on public/app/features/alerting/unified/home/GettingStarted.tsx:
```diff
@@ -38,107 +37,72 @@ export default function GettingStarted({ showWelcomeHeader }: { showWelcomeHeade
-          <div>
-            <ArrowLink href="https://grafana.com/docs/grafana/latest/alerting/" title="Read more in the Docs" />
-          </div>
+          <TextLink href="https://grafana.com/docs/grafana/latest/alerting/" icon="angle-right" inline={false}>
```
Sadly we can't use `TextLink` for external refs unless `external={true}`, that will render a proper `<a>` tag, otherwise the URL is passed in to the regular react-router-dom `<Link>` which doesn't support arbitrary http(s) hrefs :(

tomratcliffe replied:
Oops - thanks! Fixed this. 

I was going to add a test for this (to make sure that `TextLink` with an external link but no `external` prop set would throw an error), but[ there's some existing logic](https://github.com/grafana/grafana/pull/81799) to handle external URLs and stripping base URLs etc. and I thought it best not to complicate the component any further?

gillesdemey reviewed (commented):
Needs a small update to the documentation link

gillesdemey commented:
SVG doesn't seem to render on Safari; I can look in to that if you want :) 
<img width="734" alt="image" src="https://github.com/grafana/grafana/assets/868844/e5a62c2f-cc69-4eb4-bb9e-e8ed34e709d0">

tomratcliffe committed:
Fix display of Alert diagram in Safari

tomratcliffe committed:
Add missing `external` link for TextLink

tomratcliffe commented:
> SVG doesn't seem to render on Safari...

Thanks! Fixed this as well, it was relying on the `flex` to make sure it filled the container. Added back in now

gillesdemey reviewed (approved):
Changes LGTM! 🚀
